### ISSUE TITLE
Attaching particlespawners to certain entities causes segfaults

### ISSUE THREAD

**Issue by Jordach** [minetest/minetest#10063]
##### Minetest version
```
minetest-5.3.0-dev-0a1181f76
```

##### Summary
~~Running this code; https://github.com/Jordach/SolarSail/blob/f49a9209a283dd77d86e5a311255fc0d4b78edfe/mods/weapons/weapons/grenades.lua#L87-L117~~

Cause a unique segfault within `content_cao.cpp:193`

This is caused by particles.cpp:352

```cpp
attached_absolute_pos_rot_matrix = &attached->getAbsolutePosRotMatrix();
```

##### To reproduce:
Create an entity with the definition set to ```is_visible = false```
Spawn the entity and then attach a particlespawner to it.
Watch as the engine segfaults.

**Comment by LoneWolfHT** [comment]
Can confirm, I get this with my hammer_of_power mod whenever I hammer-smash near a hidden wielditem, which you can find a lot of on Nodecore players. The mod applies add_velocity to the object and attaches a particlespawner (https://github.com/LoneWolfHT/hammer_of_power/blob/master/init.lua#L213).
A test mod can be made if wanted

**Comment by pauloue** [comment]
That line should probably be:
```
attached_absolute_pos_rot_matrix = &(attached->getAbsolutePosRotMatrix());
```
Or remove the `&` and make `spawnParticle` take a reference instead of a pointer.

**Comment by SmallJoker** [comment]
@pauloue `->` is always evaluated before address-of `&`.
https://en.cppreference.com/w/cpp/language/operator_precedence
----

There's neither a scene node nor a matrix node for invisible entities. This case should really be checked there. Writing a patch for this ...

**Comment by SmallJoker** [comment]
Segfault fixed in 57df895cf
